text/0000-standard-dependency-metadata-format.md

```diff
@@ -0,0 +1,48 @@
+# Standardized Paketo Dependency Metadata Format
+
+## Summary
+
+This proposal suggests that the Paketo project should converge on a single dependency metadata format.
+
+## Motivation
+
+There are several projects, such as the introduction of ARM or the removal of stacks, that are being discussed that would force Paketo to update the metadata for dependencies. It seems prudent that if we are going to have to do a large update to dependency metadata that it might also be a good time for us to converge as a project on the dependency metadata format that we should be using going forward.
+
+This could also be a good first step in converging on shared tooling. By having a shared dependency metadata format we would have a good common convergence point to begin building universal tooling for the Paketo project. 
+
+## Implementation
+
+The following in the proposed metadata format:
+
+```toml
+[[metadata.dependencies]]
+arch = "<dependency compatible architecture>"
+checksum = "<dependency algo:checksum>"
+id = "<dependency ID>"
+os = "<dependency compatible OS>"
+uri = "<dependency URI>"
+version = "dependency version"
+
+cpes = [ "<dependency cpe>" ] #optional
+name = "<dependency name>" #optional
+purl = "<dependency purl>" #optional
+strip-components = <number of directories to strip off dependency artifact> #optional
+
+    [[metadata.dependencies.distros]] #optional
+    name = "<compatible OS distribution name>"
+    version = "<compatible OS distribution version>" #optional
+
+    [[metadata.dependencies.licenses]] #optional
+    type = "<license of dependency>"
+    uri = "<URI for information of license>" #optional
+```
+
+**Note:** Both the `distros` and `licenses` fields are optional, however if they are given then the non-optional components of them must be set.
+
+## Prior Art
+- The layout of distributions is pulled from the [Buildpacks Spec](https://github.com/buildpacks/spec/blob/main/buildpack.md#buildpacktoml-toml).
+
+## Unresolved Questions / Bikeshedding
+- Should we make a new tooling repository as part of this initial RFC or should that wait for implementation?
+- Should this include a buffer for stacks or should the stack removal be the forcing function that forces this conversion?
+- Does this require the creatation of our own Paketo Spec or some other API style document or is this RFC sufficent?
```

Comment:
I think it depends on the goal. Are we going to ever certify someone as "Paketo compliant"? If so, then we should start working towards an official spec. If not, if the intent is to guide implementation then I think the RFCs are OK. We do maintain a list of the "active" RFCs, so someone interested in compatibility can reference them.

I don't think we need to put this into the documentation until it affects normal end users. I think it's a reasonable expectation for buildpack authors to sometimes need to refer to an RFC directly.

Reply:
I think that might be quite nice to have this. It would be easy to find the current metadata format (and not bury it in some RFC) und future changes will be easy to track.

Reply:
I am not entirely sure what we should do about a "spec". It feels like something that is needed as a result of this RFC but I am not sure that this RFC is the right place to hash out what we should do and where it should live. Should I remove this as a concern for now and open a separate discussion somewhere on what we should do to document these kinds of standards within the Paketo community? I would like to try and keep this rolling forward for the time being so that we can continue to advance both the multi-arch as well as the decoupled dependencies buildpacks rfcs forward.

Reply:
+1 for splitting out a spec discussion. It's a good discussion, one we need to have. It'll take a lot of work to do right. It shouldn't hold up this or other efforts. My $0.02.

Reply:
It doesn't have to block the merge of the RFC though. And, given the "informal" description that is part of the RFC, it can be adopted while we figure out a process for writing Paketo specific specifications.